crates/client/src/api.rs

```diff
@@ -216,6 +223,31 @@ impl Client {
     pub fn url(&self) -> &RegistryUrl {
         &self.url
     }
+    /// Gets the `.well-known` configuration registry URL.
+    pub async fn well_known_config(&self) -> Result<Option<RegistryUrl>, ClientError> {
+        let url = self.url.join(WELL_KNOWN_PATH);
+        tracing::debug!(url, "getting `.well-known` config",);
+
+        let res = self.client.get(url).send().await?;
+
```

Comment:
Maybe `tracing::debug` the response status here? Or at least in the `!is_success` case below.